titan/interactions/pca.py

```diff
@@ -26,27 +26,33 @@ def interact(model: "model.HIVModel", rel: "agent.Relationship") -> bool:
         """
         if not model.params.features.pca or model.time < model.params.pca.start_time:
             return False
-
         assert (
             model.params.model.network.enable
         ), "Network must be enabled for pca interactions"
 
         params = model.params.partnership.pca.frequency[rel.bond_type]
-        acts_prob = model.run_random.random()
-        acts_bin = 1
-        current_p_value = params[acts_bin].prob
-
-        while acts_prob > current_p_value:
-            acts_bin += 1
-            current_p_value += params[acts_bin].prob
-
-        min = params[acts_bin].min
-        max = params[acts_bin].max
 
-        if min == max:
-            num_acts = min
+        if params.type == "bins":
+            params = params.bins
+            acts_prob = model.run_random.random()
+            acts_bin = 1
+            current_p_value = params[acts_bin].prob
+
+            while acts_prob > current_p_value:
+                acts_bin += 1
+                current_p_value += params[acts_bin].prob
+
+            min = params[acts_bin].min
+            max = params[acts_bin].max
+
+            if min == max:
+                num_acts = min
+            else:
+                num_acts = model.run_random.randrange(min, max)
+        elif params.type == "distribution":
+            num_acts = round(utils.safe_dist(params.distribution, model.run_random))
         else:
```

Comment:
this is technically enforced by paraml, so we could probably skip the else and let it just crash and burn if it breaks (by someone reaching into the params and manually changing things)

Reply:
Want to do that on your branch since that implements paraml?

Reply:
by param, I mean generally our params parsing, it should already be in place on this branch